### PR TITLE
executor: unstable test TestFlushTables move to standalone sandbox

### PR DESCRIPTION
**Pull request by hawkingrei**
Signed-off-by: Weizhen Wang <wangweizhen@pingcap.com>

### What problem does this PR solve?

Issue Number: ref #35247

Problem Summary:

### What is changed and how it works?

```executor``` package contains too many tests to use sandbox by Bazel. Bazel allows users to use 50 shares to run the test case in the sandbox. Otherwise, Bazel thinks the organization of the test case is too chaotic and asks users to reorg the test case.

So we need to put some test cases into subcategories to make it isolate.

### Check List

Tests 

- [x] Unit test

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by tisonkun — COMMENTED** [review]
Can this change fix the "unstable" issue? Or just fit what bazel suggests?

**Comment by hawkingrei** [comment]
> Can this change fix the "unstable" issue? Or just fit what bazel suggests?

Both. I 

> Can this change fix the "unstable" issue? Or just fit what bazel suggests?

It just put it into an isolated sandbox and it also fits with bazel suggests.

**Review by tisonkun — APPROVED** [review]

File: executor/simpletest/main_test.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+	tikv.EnableFailpoints()
+
+	opts := []goleak.Option{
+		goleak.IgnoreTopFunction("github.com/golang/glog.(*loggingT).flushDaemon"),
+		goleak.IgnoreTopFunction("go.etcd.io/etcd/client/pkg/v3/logutil.(*MergeLogger).outputLoop"),
+		goleak.IgnoreTopFunction("go.opencensus.io/stats/view.(*worker).start"),
+		goleak.IgnoreTopFunction("gopkg.in/natefinch/lumberjack%2ev2.(*Logger).millRun"),
+		goleak.IgnoreTopFunction("github.com/tikv/client-go/v2/txnkv/transaction.keepAlive"),
+	}
+	goleak.VerifyTestMain(m, opts...)
```
Comment (tisonkun) [inline executor/simpletest/main_test.go:43]: I suggest you verify whether we should still need all these settings.

Comment (hawkingrei) [reply executor/simpletest/main_test.go:43]: Great, I leave two config item.

File: executor/simpletest/main_test.go [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+func TestMain(m *testing.M) {
+	autoid.SetStep(5000)
+	config.UpdateGlobal(func(conf *config.Config) {
+		conf.Log.SlowThreshold = 30000 // 30s
+		conf.TiKVClient.AsyncCommit.SafeWindow = 0
+		conf.TiKVClient.AsyncCommit.AllowedClockDrift = 0
+		conf.Experimental.AllowsExpressionIndex = true
+	})
+	tikv.EnableFailpoints()
```
Comment (tisonkun) [inline executor/simpletest/main_test.go:34]: Perhaps these lines can be removed also. At least I don't see a dependency to tikv failpoints.

Comment (hawkingrei) [reply executor/simpletest/main_test.go:34]: Done

**Review by xhebox — APPROVED** [review]

**Pull request merged**